Client/core/Graphics/CGraphics.cpp

```diff
@@ -668,18 +679,11 @@ float CGraphics::GetDXTextExtentW(const wchar_t* wszText, float fScale, LPD3DXFO
     {
         // DT_CALCRECT may not take space characters at the end of a line into consideration for the rect size.
         // Count the amount of space characters at the end
-        size_t       spaceCount = 0;
         const size_t textLength = wcslen(wszText);
 
-        for (int i = textLength - 1; i >= 0; --i)
-        {
-            const wchar_t c = wszText[i];
-
-            if (c == L' ')
-                ++spaceCount;
-            else
-                break;
-        }
```

Comment:
Old code is clearer.

[No difference in performance.](http://quick-bench.com/FusJ6BGP4JbWPbnZp_03oH5Az98)

<details><summary>Code used, in case quick-bench.com goes down</summary>

```cpp
#include <string.h>

static void Old(benchmark::State& state) {
  // Code inside this loop is measured repeatedly
  for (auto _ : state) {
    const char* x = "hi world!";
    const int textLength = strlen(x);
    int spaceCount = 0;
    for (int i = textLength - 1; i >= 0; --i)	{
      const char c = x[i];
      if (c == ' ')
        ++spaceCount;
      else
        break;
    }
    // Make sure the variable is not optimized away by compiler
    benchmark::DoNotOptimize(x);
    benchmark::DoNotOptimize(spaceCount);
  }
}
// Register the function as a benchmark
BENCHMARK(Old);

static void New(benchmark::State& state) {
  // Code inside this loop is measured repeatedly
  for (auto _ : state) {
    const char* x = "hi world!";
    const int textLength = strlen(x);
    int spaceCount = 0;
    for (int i = textLength - 1; i >= 0 && x[i] == L' '; --i)
      spaceCount++;
    
   
    // Make sure the variable is not optimized away by compiler
    benchmark::DoNotOptimize(x);
    benchmark::DoNotOptimize(spaceCount);
  }
}
BENCHMARK(New);
```

</details>

Reply:
The compiler optimized it away, I knew it, i just thought the new way would be cleaner.
But okay, no problem.

Reply:
im just so stuck with this one liner..
```cpp
while(wszText[textLength - 1 - spaceCount++] != L' ');
```
this is my new, totally not confusing method. It has the same efficiency as the others. But alright, I'll leave the current for loop version. I agree its kinda cleaner.